### PR TITLE
fixed detection_based_tracker.cpp on visual studio 2013 and later

### LINKED ISSUE #5023: DetectionBasedTracker
Transferred from http://code.opencv.org/issues/4410

```
|| Bruno Nascimento on 2015-06-15 10:56
|| Priority: Normal
|| Affected: branch 'master' (3.0-dev)
|| Category: None
|| Tracker: Bug
|| Difficulty: 
|| PR: 
|| Platform: x64 / Windows
```
## DetectionBasedTracker

```
Is there any reason why DetectionBasedTracker sample is only for LINUX and ANDROID?

Im also getting linker problems when I try to use it on windows, on another project.

Error   1   error LNK2019: unresolved external symbol "public: __thiscall cv::DetectionBasedTracker::Parameters::Parameters(void)" (??0Parameters@DetectionBasedTracker@cv@@QAE@XZ) referenced in function _main

When I try to use it on windows:
cv::DetectionBasedTracker::Parameters params;
cv::DetectionBasedTracker Detector( mainDetector, trackingDetector, params );
if ( !Detector.run() )
{
    cerr << "Detector initialization failed" << endl;
    delete detector;
    exit( EXIT_FAILURE );
}
```
##### Alexander Alekhin on 2015-06-15 11:07

```
DetectionBasedTracker should be available in MSVS2012/MSVS2013 builds and it is not available in MSVS2010 during hard dependency to std::thread ( C++11 features: http://en.cppreference.com/w/cpp/thread/thread )
```
##### Bruno Nascimento on 2015-06-15 11:18

```
Im actually using MSVS2013, I only have this problem on x86 Debug mode. My x86 Release mode does not have this linker problem since its build as static.
Both my Debug and Release modes on x64, are also showing this kind on linker problems when using DetectionbasedTracker.
e.g.

Error   1   error LNK2001: unresolved external symbol "public: __cdecl cv::DetectionBasedTracker::Parameters::Parameters(void)" (??0Parameters@DetectionBasedTracker@cv@@QEAA@XZ)   C:\Users\bnascimento\Projects\Tracking\main.obj Tracking
```
##### Alexander Alekhin on 2015-06-15 12:15

```
Thanks for details update!

Implementation of DetectionBasedTracker is broken (nested class DetectionBasedTracker::Parameters is not exported).
Sample "dbt_face_detection.cpp" doesn't support Windows.
```

### PR DESCRIPTION
**Pull request by es0m**
### This pullrequest removes the shared unique_lock<mutex> in cv::DetectionBasedTracker::SeparateDetectionWork and moves the default constructor into the header file.

Reusing the lock across multiple threads resulted in std::system_error (actually standard conformant). 
Basically, the clue is in the name. Unique locks are unique locks (to phrase it with Theresa May). 

closes https://github.com/opencv/opencv/issues/5023

### CODE REVIEW

**Comment by alalek** [comment]
Some files have attribute changes:

> 100644 → 100755

Please remove executable bit from source files.

**Comment by es0m** [comment]
How come my executable flag change breaks the stitching performance test on Windows 10 x64: http://pullrequest.opencv.org/buildbot/builders/precommit_windows_ten/builds/3219

**Comment by es0m** [comment]
I squashed the commit messages to trigger another build and now the opencl perf tests fail but the previous one works: http://pullrequest.opencv.org/buildbot/builders/precommit_opencl/builds/7517

File: samples/cpp/dbt_face_detection.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#if defined(__linux__) || defined(LINUX) || defined(__APPLE__) || defined(ANDROID)
+#if defined(__linux__) || defined(LINUX) || defined(__APPLE__) || defined(ANDROID) || (defined(_MSC_VER) && _MSC_VER>1700)
```
Comment (alalek) [inline samples/cpp/dbt_face_detection.cpp]: `>1770` -> `>=1700`

Comment (es0m) [inline samples/cpp/dbt_face_detection.cpp]: actually, >1700 was right, unless there is a release between VS2012 and VS2013. VS2013 is 1800. So now it's clearer.

Comment (alalek) [inline samples/cpp/dbt_face_detection.cpp]: BTW, Header file has check for >=1700. What VS2013 feature is required in sample?

Comment (es0m) [inline samples/cpp/dbt_face_detection.cpp]: I tried to dig this out ( https://msdn.microsoft.com/en-us/library/hh567368.aspx#concurrencytable ) but can't find anything certain. TBH, the code was running on 2013 with minor modifications, but on 2015 it was throwing system_error according to standard. 
I don't have 2012 available so cannot confirm either compilation nor functionality.

**Comment by vpisarev** [comment]
thank you! :+1:

**Pull request merged**